src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -1041,64 +1016,43 @@ private ErrorRecord CreateModuleNotFoundError(string modulePath)
                 }
             }
 
-            return modules;
+            // Make sure we always return a non-null collection.
+            return allModules ?? Utils.EmptyArray<PSModuleInfo>();
         }
 
         /// <summary>
-        /// Get a list of all modules
-        /// which can be imported just by specifying a non rooted file name of the module
-        /// (Import-Module foo\bar.psm1;  but not Import-Module .\foo\bar.psm1)
+        /// Get modules based on the given names and module files.
         /// </summary>
-        private IEnumerable<PSModuleInfo> GetAllAvailableModules(string directory, bool refresh)
+        private IEnumerable<PSModuleInfo> GetModulesFromOneModulePath(string[] names, string modulePath, bool all, bool refresh)
```

Comment:
How hard is the requirement on `names` being an array? I only ask because I notice this part of the code tends to have collection types in signatures vary a lot (some `IEnumerable<string>`, some `string[]`, some `List<string>`) and we should probably work out a better rationale for what types we prefer.

If there is a particular performance implication from using `string[]` then that makes sense, but in general I think we should favour the principle of using an interface with only the properties required by each method; if we only want to go through entries sequentially, `IEnumerable<T>`, if we want to know how many there are up front, something like `ICollection<T>`, if we want indexing `IList<T>`.

I'm not making a particular suggestion here I suppose. I just have the impression that the various collection types used in the methods in the module cmdlets weren't used for any particular reason, and we should be more judicious about what collection types we use (and the semantics of each) given how intensively we handle them.

Reply:
For this method, it makes more sense to use `List<string>`, so the top-level caller doesn't need to cast the List to a string array.